[DOCS] Adding a docker compose version that includes the kitchen sink (#9688)
apache/pulsar: docker-compose/kitchen-sink/README.MD

<!--

    Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

-->
# Apache Pulsar Docker Compose

> In this folder path run the following command to start the cluster

Reviewer: ```suggestion
In this folder path, run the following command to start the cluster.
```

```
docker-compose up
```

> This will startup individual containers for the following services:

Reviewer: ```suggestion
This will start individual containers for the following services:
```

> - zookeeper (3)

Reviewer: ```suggestion
- ZooKeeper (3)
```

> - bookkeeper (3)

Reviewer: ```suggestion
- BookKeeper (3)
```

> - pulsar (3)

Reviewer: ```suggestion
- Pulsar (3)
```

> - proxy (1)

Reviewer: ```suggestion
- Proxy (1)
```

> - websocket (1)

Reviewer: ```suggestion
- WebSocket (1)
```

> - function (1)

Reviewer: ```suggestion
- Function (1)
```

> - pulsar manager (1)

Reviewer: ```suggestion
- Pulsar Manager (1)
```

> - sql (1)

Reviewer: ```suggestion
- SQL (1)
```

## Motivation

> Most of what is available with Docker and Apache Pulsar used the standalone version of Apache Pulsar, which is fantastic, but takes care of all of the complexity of Apache Pulsar.

Reviewer: ```suggestion
Most of what is available with Docker and Apache Pulsar uses the standalone version of Apache Pulsar, which is fantastic. But, users need to take care of all of the complexity of Apache Pulsar.
```

This was created to create a Docker implementation that allows each individual part to be exposed to learn how it all fits together.

> ## Urls

Reviewer: ```suggestion
## URLs
```

Assuming localhost the following endpoint will be available:

> [web-dashboard - http://localhost:9527](http://localhost:9527) pulsar admin dashboard, shows various metrics and metatdata information about the cluster

Reviewer: ```suggestion
[web-dashboard - http://localhost:9527](http://localhost:9527) pulsar-admin dashboard, showing various metrics and metadata information about the cluster
```

> [broker-admin - http://localhost:8080](http://localhost:8080) access broker rest  interface

Reviewer: ```suggestion
[broker-admin - http://localhost:8080](http://localhost:8080) access broker rest interface
```

> [broker-service-url - pulsar//:locahost:6650](pulsar//:locahost:6650)  broker service url for use with producers and consumers

Reviewer: ```suggestion
[broker-service-url - pulsar//:locahost:6650](pulsar//:locahost:6650)  broker service URL for use with producers and consumers
```

You can shut it down with the following command in folder path:

```
docker-compose down
```

> You may need to delete the `./data` folder created by the pulsar manager.

Reviewer: ```suggestion
You may need to delete the `./data` folder created by the Pulsar Manager.
```

## Notes

Docker is notorious for being difficult to manage the startup order of containers. In Pulsar, startup order matters. This may cause some containers to fail and have to be restarted.

Generally speaking, everything should be started up successfully in about 2 minutes.

If you plan to use this in production, more care should be taken on restart policies, logging, configuration, etc.

> ## TODO

Reviewer: ```suggestion
## To-Do
```

> - Add more SQL workers

Reviewer: ```suggestion
- Add more SQL workers.
```

> - Dig into why PF_configurationStoreServers in fnc1 does not support a list of zookeeper instances

Reviewer: ```suggestion
- Dig into why PF_configurationStoreServers in fnc1 does not support a list of ZooKeeper instances.
```

> - Figure out why v0.2.0 of the dashboard was not working correctly

Reviewer: ```suggestion
- Figure out why the dashboard ( v0.2.0) does not work correctly.
```
